conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/Enclave.kt

```diff
@@ -212,30 +233,20 @@ abstract class Enclave {
 
     @Suppress("unused")  // Accessed via reflection
     @PotentialPackagePrivate
-    private fun initialise(env: EnclaveEnvironment, upstream: Sender): HandlerConnected<*> {
+    private fun initialise(env: EnclaveEnvironment) {
         this.env = env
-        // If the Enclave class implements InternalEnclave then the behaviour of the enclave is entirely delegated
-        // to the InternalEnclave implementation and the Conclave-specific APIs (e.g. callUntrustedHost, etc) are
-        // disabled. This allows us to test the enclave environment in scenarios where we don't want the Conclave handlers.
-        return if (this is InternalEnclave) {
-            this.internalInitialise(env, upstream)
-        } else {
-            initCryptography()
-            /*
-            Here we should not add any code that throws exceptions. This is because the handler mechanisms haven't
-            yet been put in place and an exception would cause the Enclave to terminate in such a way that we can't
-            control and without meaningful stacktrace as this can't be reported back.
-            */
-            val exceptionSendingHandler = ExceptionSendingHandler(env.enclaveMode == EnclaveMode.RELEASE)
-            val connected = HandlerConnected.connect(exceptionSendingHandler, upstream)
-            val mux = connected.connection.setDownstream(SimpleMuxingHandler())
-            adminHandler = mux.addDownstream(AdminHandler(this, env))
-            quotingEnclaveInfoHandler = mux.addDownstream(QuotingEnclaveInfoHandler())
-            signedQuoteHandler = mux.addDownstream(SignedQuoteHandler())
-            enclaveInstanceInfoQuoteHandler = mux.addDownstream(EnclaveInstanceInfoQuoteHandler())
-            enclaveMessageHandler = mux.addDownstream(EnclaveMessageHandler())
-            connected
+        initCryptography()
+
+        env.hostInterface.apply {
```

Comment:
Can these be registered at creation time of the enclave object, i.e. inside an `init`?

Reply:
Possibly, but not easily.